Detour/Include/DetourCommon.h

```diff
@@ -482,6 +483,15 @@ inline void dtSwapEndian(float* v)
 void dtRandomPointInConvexPoly(const float* pts, const int npts, float* areas,
 							   const float s, const float t, float* out);
 
+template<typename TypeToRetrieveAs, typename BufferType>
+TypeToRetrieveAs* getThenAdvanceBufferPointer(BufferType*& buffer, const size_t distanceToAdvance)
```

Comment:
Other global Detour functions are prefixed by `dt`. I think we should do the same here.
Does `buffer` need to be a templated type?